src/core/core.cpp

```diff
@@ -0,0 +1,62 @@
+#include <vector>
+#include <nlohmann/json.hpp>
+#include <string>
+
+using json = nlohmann::json;
+
+
+struct Product {
+    std::string name;
+    int amount;
+    double price;
```

Comment:
read about fields padding

https://www.geeksforgeeks.org/structure-member-alignment-padding-and-data-packing/

when you sort the fields from largest to smallest, the size of the structure in memory will be smaller
This is not important when you programming applications that have unlimited memory resources, but in embedded programming, e.g. of a processor, where we have memory limits, the arrangement of fields in the structure may be very important.


struct Product {
    std::string name;
    double price;
    int amount;
    int whoBoughtID;
};